cmd/snapshot/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"image"
+	"image/png"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/go-spatial/geom"
+	"github.com/go-spatial/geom/slippy"
+	mbgl "github.com/go-spatial/go-mbgl"
+)
+
+/*
+snapshotter bounds "lat long lat long" -width=100, -height=100 style filename.png
+snapshotter tile "z/x/y" style filename.png
+*/
+
+type cmdType uint8
+
+const (
+	CmdUnknown = cmdType(iota)
+	CmdBounds
+	CmdTile
+)
+
+func (c cmdType) String() string {
+	switch c {
+	case CmdBounds:
+		return "Bounds"
+	case CmdTile:
+		return "Tile"
+	default:
+		return "Unknown"
+	}
+}
+
+var FWidth uint
+var FHeight uint
+var FStyle string
+var FPixelRatio float64
+
+var FBounds geom.Extent
+var FTile slippy.Tile
+var FOutputFilename string
+
+func usage() {
+	fmt.Fprintf(os.Stderr, "%v [options...] \"lat long lat long\" output_filename.png\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "%v [options...] bounds \"lat long lat long\" output_filename.png\n", os.Args[0])
+	fmt.Fprintf(os.Stderr, "%v [options...] tile \"z/x/y\" output_filename.png\n", os.Args[0])
+	flag.PrintDefaults()
+}
+
+func parseBounds(boundString string) {
+	var err error
+	bounds := strings.Split(boundString, " ")
+	if len(bounds) != 4 {
+		fmt.Fprintf(os.Stderr, "Error: invalid bounds provided — %v\n", boundString)
+		usage()
+		os.Exit(2)
+	}
+	for i, bound := range bounds {
+		FBounds[i], err = strconv.ParseFloat(strings.TrimSpace(bound), 64)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "Error: invalid bounds provided — %v\n", boundString)
+			fmt.Fprintf(os.Stderr, "Error: unabled to parse %v(%v) as a float.\n", bound, i)
+			usage()
+			os.Exit(2)
+		}
+	}
+}
+
+func parseTile(tileString string) {
+	var err error
+	var v uint64
+	parts := strings.Split(tileString, "/")
+	if len(parts) != 3 {
+		fmt.Fprintf(os.Stderr, "Error: invalid z/x/y coordinates  — %v\n", tileString)
+		usage()
+		os.Exit(2)
+	}
+	var label = [...]string{"Z", "X", "Y"}
+
+	for i, part := range parts {
+		switch i {
+		case 0:
+			v, err = strconv.ParseUint(strings.TrimSpace(part), 10, 64)
```

Comment:
Could this be outside of the switch statement?

Reply:
yes.